tests/Api/Shop/PayingForOrdersByVisaMobileTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Api\Shop;
+
+use CommerceWeavers\SyliusTpayPlugin\Tpay\PayGroup;
+use Sylius\Component\Core\Model\OrderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Tests\CommerceWeavers\SyliusTpayPlugin\Api\JsonApiTestCase;
+use Tests\CommerceWeavers\SyliusTpayPlugin\Api\Utils\CardEncrypterTrait;
+use Tests\CommerceWeavers\SyliusTpayPlugin\Api\Utils\OrderPlacerTrait;
+
+class PayingForOrdersByVisaMobileTest extends JsonApiTestCase
+{
+    use CardEncrypterTrait;
+    use OrderPlacerTrait;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpOrderPlacer();
+    }
+
+    public function test_paying_with_visa_mobile_based_payment_type(): void
+    {
+        $this->loadFixturesFromDirectory('shop/paying_for_orders_by_card');
+
+        $order = $this->doPlaceOrder('t0k3n');
+
+        $this->client->request(
+            Request::METHOD_POST,
+            sprintf('/api/v2/shop/orders/%s/pay', $order->getTokenValue()),
+            server: self::CONTENT_TYPE_HEADER,
+            content: json_encode([
+                'successUrl' => 'https://example.com/success',
+                'failureUrl' => 'https://example.com/failure',
+                'payer' => [
+                    'phone' => '44123456789',
+                ],
+                'pay' => [
+                    'groupId' => PayGroup::VISA_MOBILE,
+                ],
+            ]),
+        );
+
+        $response = $this->client->getResponse();
+
+        $this->assertResponseCode($response, Response::HTTP_OK);
+        $this->assertResponse($response, 'shop/paying_for_orders_by_visa_mobile/test_paying_with_visa_mobile_payment_type');
+    }
+
+    private function doPlaceOrder(
```

Comment:
Why can't we change the definition of such a method in OrderPlacerTrait to protected?

Reply:
So why do we duplicate its content here?